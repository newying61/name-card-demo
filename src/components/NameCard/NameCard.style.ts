import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  border: 1px solid #ddd;
  border-radius: 5px;
  width: 70%;
  min-width: 12.5rem;
  margin: auto;
  padding: 1rem;
  margin: 1rem;

  @media only screen and (min-width: 600px) {
    width: 40%;
    max-width: 14.5rem;
  }
`;
