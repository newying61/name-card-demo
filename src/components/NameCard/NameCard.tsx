import React from 'react';
import { NameCard as NameCardType } from '../../types/NameCardType';
import NameCardItem from '../NameCardItem';
import ListGroup from 'react-bootstrap/ListGroup';
import { StyledButton, ButtonsWrapper } from '../Styled/Buttons';
import { Wrapper } from './NameCard.style';
import { FieldNameMap } from '../../constants';

interface NameCardProps {
  card: NameCardType,
  onEdit: (card: NameCardType) => void,
  onDelete: (card: NameCardType) => void
}

const NameCard: React.FC<NameCardProps> = ({ card, onEdit, onDelete }) => {
  const onEditBtnClick = () => {
    onEdit(card);
  }

  const onDeleteBtnClick = () => {
    onDelete(card);
  }

  return (
    <Wrapper>
      <ListGroup as="ul" variant="flush">
        {Object.keys(card).map((c) => {
          if (c === 'id') {
            return null;
          }

          return (
            <NameCardItem key={c} fieldName={FieldNameMap[c]} fieldValue={card[c]} />
          );
        })}
      </ListGroup>
      <ButtonsWrapper>
        <StyledButton variant="primary" onClick={onEditBtnClick}>Edit</StyledButton>
        <StyledButton variant="danger" onClick={onDeleteBtnClick}>Delete</StyledButton>
      </ButtonsWrapper>
    </Wrapper>
  );
}

export default NameCard;