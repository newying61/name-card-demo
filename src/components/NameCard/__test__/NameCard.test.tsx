import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import NameCard from '../';

afterEach(cleanup);

describe('AppNav', () => {
  it('should render without crash', () => {
    const card = {
        id: 'test',
        firstName: 'test',
        lastName: 'test',
        dateOfBirth: '2019-01-01'
    };

    const onEdit = jest.fn();
    const onDelete = jest.fn();

    const { container, getByText } = render(
      <NameCard card={card} onEdit={onEdit} onDelete={onDelete} />
    );

    fireEvent.click(getByText('Edit'));
    expect(onEdit).toBeCalled();

    fireEvent.click(getByText('Delete'));
    expect(onEdit).toBeCalled();

    expect(container.firstChild).toMatchSnapshot();
  });
});