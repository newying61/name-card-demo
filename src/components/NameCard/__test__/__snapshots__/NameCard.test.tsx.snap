// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav should render without crash 1`] = `
<div
  class="sc-ifAKCX plnT"
>
  <ul
    class="list-group list-group-flush"
  >
    <li
      class="list-group-item"
    >
      <span
        class="sc-bdVaJa dlUoII"
      >
        First name
      </span>
      <span
        class="sc-bwzfXH gaDNSi"
      >
        test
      </span>
    </li>
    <li
      class="list-group-item"
    >
      <span
        class="sc-bdVaJa dlUoII"
      >
        Last name
      </span>
      <span
        class="sc-bwzfXH gaDNSi"
      >
        test
      </span>
    </li>
    <li
      class="list-group-item"
    >
      <span
        class="sc-bdVaJa dlUoII"
      >
        Date of birrth
      </span>
      <span
        class="sc-bwzfXH gaDNSi"
      >
        2019-01-01
      </span>
    </li>
  </ul>
  <div
    class="sc-bxivhb jcdCai"
  >
    <button
      class="sc-htpNat eZpMOu btn btn-primary"
      type="button"
    >
      Edit
    </button>
    <button
      class="sc-htpNat eZpMOu btn btn-danger"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
